source/components/sidebar/sidebar-component.js

```diff
@@ -48,8 +53,51 @@ class Sidebar extends PureComponent {
     this.props.onLogoClick();
   }
 
+  toggleSubmenu(index) {
+    const openedAccordion = this.state.openedAccordion === index ? null : index;
+    this.setState({ openedAccordion });
+  }
+
+  getActive(index) {
+    return index === this.state.openedAccordion;
+  }
+
+  renderMenuWithAccordion({ icon, name, submenu }, index) {
+    const { onClickItem } = this.props;
+    return (
+      <MenuItem active={this.getActive(index)}>
+        <AccordionTitle
+          active={this.getActive(index)}
+          index={index}
+          onClick={() => this.toggleSubmenu(index)}
+          icon={icon}
+        >
+          {name}
+        </AccordionTitle>
+        <AccordionContent active={this.getActive(index)}>
+          <Menu>
+            {submenu.map(subitem => (
+              <MenuItem active={subitem.isActive} link={subitem.link} handleClick={onClickItem}>
```

Comment:
Adicionei essa função de handleClick, para disparar para o pai quando um link for clicado

Reply:
Talvez não fosse melhor que cada "link" e "seção" tivessem um id e esse id ficasse no estado como "active" e isso pudesse ser recebido por props? Tipo `openSection` e `activeItem`. Porque tendo que olhar o schema pra saber quem tá aberto vai precisar re-renderizar tudo quando mudar só um item, acredito.